airflow/hooks/dbapi_hook.py

```diff
@@ -62,15 +63,29 @@ def __init__(self, *args, **kwargs):
             raise AirflowException("conn_name_attr is not defined")
         elif len(args) == 1:
             setattr(self, self.conn_name_attr, args[0])
-        elif self.conn_name_attr not in kwargs:
-            setattr(self, self.conn_name_attr, self.default_conn_name)
         else:
-            setattr(self, self.conn_name_attr, kwargs[self.conn_name_attr])
+            setattr(self, self.conn_name_attr, kwargs.get(self.conn_name_attr, self.default_conn_name))
+
+    @property
+    def conn_name(self) -> str:
+        """
+        Get the name of the database connection attribute which identifies the connection.
+        """
+        return getattr(self, self.conn_name_attr)
+
+    @property
+    def connection(self) -> Connection:
+        """
+        Get the database connection object.
+
+        :return: Connection
+        """
+        return super().get_connection(self.conn_name)
```

Comment:
```suggestion
        return self.get_connection(self.conn_name)
```
should also work, doesn't it?

Actually I think this option (the second one) is better. 👍

Reply:
Haha, I think so, and I open this to hear other committer idea